src/WebJobs.Script.WebHost/Diagnostics/UserLogMetricsLoggerProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class UserLogMetricsLoggerProvider : ILoggerProvider, ISupportExternalScope
+    {
+        private readonly IMetricsLogger _metricsLogger;
+        private IExternalScopeProvider _scopeProvider;
+
+        public UserLogMetricsLoggerProvider(IMetricsLogger metricsLogger)
+        {
+            _metricsLogger = metricsLogger;
+        }
+
+        public ILogger CreateLogger(string categoryName)
+        {
+            return new UserLogMetricsLogger(categoryName, _metricsLogger, _scopeProvider);
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public void SetScopeProvider(IExternalScopeProvider scopeProvider)
+        {
+            _scopeProvider = scopeProvider;
+        }
```

Comment:
`ISupportExternalScope` and `IExternalScopeProvider` are newer interfaces that we've not used previously. They allow us to share a logging scope and do it in a common way. This means we can remove a lot of our custom `DictionaryScope` stuff. I'll do that in a follow-up PR, though.